grammars/mlir.json

```diff
@@ -1,112 +1,590 @@
+    "vectorTransfer":{
+      "match" : "\\Wvector\\.transfer_(read|write)\\b",
+      "name" : "entity.name.function.mlir.vectorTransfer"
+    },
+    "vectorTuple":{
+      "match" : "\\Wvector\\.(tuple|tuple_get)\\b",
+      "name" : "entity.name.function.mlir.vectorTuple"
+    },
+    "loopOps":{
+      "match" : "\\bloop\\.(for|if|parallel|reduce|yield)\\b",
+      "name": "entity.name.function.mlir.loopOps"
+    },
+    "tileFunctions":{
+      "match" : "\\btile\\.(constant|contract|index|reshape)\\b",
+      "name": "entity.name.function.mlir.tileFunctions"
+    },
+    "tileQuotedFunctions":{
+      "match" : "(?<=\\W)\"tile\\.(constant|contract|index|reshape)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tileFunctions"
+    },
+    "tileKeywords": {
+      "match": "\\btile\\.(name)\\b",
+      "name": "keyword.other.mlir.tileKeywords"
+    },
+    "eltwiseFunctions":{
+      "match" : "(?<=\\W)\"eltwise\\.(add|div|sconst|cmp_lt|select|sub|exp|cast|neg|mul|sqrt|ident)\"(?=\\W)",
+      "name": "entity.name.function.mlir.tileFunctions"
+    },
+    "CHECK": {
+      "match": "(\/\/)\\s*(CHECK\\s*:|CHECK-\\w+\\s*:)(.*)$",
+      "captures": {
+        "1": {
+          "name": "comment.line.double-slash.mlir"
+        },
+        "2": {
+          "name": "comment.other.CHECK.mlir"
+        },
+        "3": {
+          "patterns": [
+            {
+              "include": "#CHECK-CODE"
+            }
+          ]
+        },
+        "name": "comment.other.CHECK.mlir"
+      }
+    },
+    "CHECK-CODE": {
+      "match": "([^0-9a-zA-Z%\"@]*)?([0-9a-zA-Z\"\\.%_\\-@]*)([^0-9a-zA-Z\"\\.%_\\-].*)?$",
+      "captures": {
+        "1": {
+          "name": "comment.line.double-slash.mlir"
+        },
+        "2": {
+          "patterns": [
+            {
+              "include": "$self"
+            },
+            {
+              "match": "(\\b|x)(index|i[1-9][0-9]*|f16|bf16|f32|f64|u8|memref|tensor|vector|func)\\b",
+              "captures": {
+                "1": {
+                  "name": "comment.line.double-slash.mlir"
+                },
+                "2": {
+                  "name": "storage.type.mlir"
+                }
+              }
+            },
+            {
+              "match": "\\b([0-9]+)(x.*)?",
+              "captures": {
+                "1": {
+                  "name": "constant.numeric.mlir"
+                },
+                "2": {
+                  "patterns": [
+                    {
+                      "include": "$self"
+                    }
+                  ]
+                }
+              }
+            },
+            {
+              "match": "(.*)",
+              "name": "comment.line.double-slash.mlir"
+            }
+          ]
+        },
+        "3": {
+          "patterns": [
+            {
+              "include": "#CHECK-CODE"
+            }
+          ]
+        }
+      }
     }
   },
-  "patterns":[
+  "patterns": [
     {
-      "include":"#comment"
```

Comment:
I find highlighting comments distracting and will be more fragile (comments are not intended to be well formed IR). Also an op in a comment is not an op, it is just a comment for the user, highlighting inside a comment would indicate these are the same semantically, while they are not.

Now an exception would be if we tried to mark code segments inside comments (e.g.,

// ```mlir
// %10 = addi %0, %1 : i32
// ```

but that would be pretty cumbersome to do in the syntax and the utility for the complexity is uncertain).